pkg/resources/saml2_integration.go

```diff
@@ -0,0 +1,891 @@
+				return property.Name == "SAML2_SIGN_REQUEST"
+			})
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("failed to find saml2 sign request, err = %w", err))
+			}
+
+			requestedNameIdFormat, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "SAML2_REQUESTED_NAMEID_FORMAT"
+			})
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("failed to find saml2 requested nameid format, err = %w", err))
+			}
+
+			forceAuthn, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "SAML2_FORCE_AUTHN"
+			})
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("failed to find saml2 force authn, err = %w", err))
+			}
+
+			snowflakeX509Cert, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "SAML2_SNOWFLAKE_X509_CERT"
+			})
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("failed to find saml2 snowflake x509 cert, err = %w", err))
+			}
+
+			snowflakeIssuerUrl, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "SAML2_SNOWFLAKE_ISSUER_URL"
+			})
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("failed to find saml2 snowflake issuer url, err = %w", err))
+			}
+
+			snowflakeAcsUrl, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "SAML2_SNOWFLAKE_ACS_URL"
+			})
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("failed to find saml2 snowflake acs url, err = %w", err))
+			}
+
+			if err = handleExternalChangesToObjectInDescribe(d,
+				describeMapping{"saml2_enable_sp_initiated", "saml2_enable_sp_initiated", enableSpInitiated.Value, enableSpInitiated.Value, nil},
+				describeMapping{"saml2_sign_request", "saml2_sign_request", signRequest.Value, signRequest.Value, nil},
+				describeMapping{"saml2_requested_nameid_format", "saml2_requested_nameid_format", requestedNameIdFormat.Value, requestedNameIdFormat.Value, nil},
+				describeMapping{"saml2_force_authn", "saml2_force_authn", forceAuthn.Value, forceAuthn.Value, nil},
+				describeMapping{"saml2_snowflake_acs_url", "saml2_snowflake_acs_url", snowflakeAcsUrl.Value, snowflakeAcsUrl.Value, nil},
+				describeMapping{"saml2_snowflake_issuer_url", "saml2_snowflake_issuer_url", snowflakeIssuerUrl.Value, snowflakeIssuerUrl.Value, nil},
+				describeMapping{"saml2_snowflake_x509_cert", "saml2_snowflake_x509_cert", snowflakeX509Cert.Value, snowflakeX509Cert.Value, nil},
+			); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+
+		// These are all identity sets, needed for the case where:
+		// - previous config was empty (therefore Snowflake defaults had been used)
+		// - new config have the same values that are already in SF
+		if !d.GetRawConfig().IsNull() {
```

Comment:
this can be also made a bit nicer after #2900

Reply:
Addressed in https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2912